frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -294,8 +294,10 @@ private void trackCurrentBranchChange(GitRepository repository) {
         if (unmanagedBranchNotification != null) {
           unmanagedBranchNotification.expire();
         }
-        if (gitMacheteRepositorySnapshot != null) {
-          val entry = gitMacheteRepositorySnapshot.getBranchLayout().getEntryByName(repositoryCurrentBranchName);
+        val snapshot = gitMacheteRepositorySnapshot;
+        val mainGitDirectory = GitVfsUtils.getMainGitDirectory(repository);
+        if (snapshot != null && snapshot.getMainGitDirectoryPath().equals(mainGitDirectory)) {
```

Comment:
Pls explain why these check is needed (so that it's obvious even outside of the context of this issue/PR)